File: webapp/src/services/UserService.ts
import { Office } from "./OfficeService";
import restClient from "./RestClient";

export class Role {
    constructor(public id: string, public name: string) {}
}

export class UserRole {
    constructor(public id: string, public role: Role) {}
}

export class User {
    constructor(
        public id: string,
        public office: Office, 
        public login: string, 
        public roles: UserRole[]) {}
}

export class UserService {
    public async getUsers(officeIds: string[]): Promise<User[]> {        
        let ids = ''
        officeIds.forEach(element => {
            ids += element + ','
        });

        let items = await restClient.post<User[]>("users/getUsers", {officeIds})
        return items
    }
}

let userService = new UserService()
export default userService

File: server/App.API/App.API/Services/UserService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using App.API.Data;
using App.API.Models;
using Microsoft.EntityFrameworkCore;

namespace App.API.Services
{
    public interface IUserService
    {
        Task<UserDto[]> GetUsersByOfficeIds(Guid[] officeIds, CancellationToken ctoken);
    }

    public class UserService : IUserService
    {
        private readonly SampleAppContext context;

        public UserService(SampleAppContext context)
        {
            this.context = context;
        }

        public async Task<UserDto[]> GetUsersByOfficeIds(Guid[] officeIds, CancellationToken ctoken)
        {
            var result =
                (from p in context.Users.Include(x => x.Office)
                    where officeIds.Contains(p.Office.Id)
                    select new UserDto()
                    {
                        Id = p.Id, 
                        Login = p.Login, 
                        Office = p.Office, 
                        Roles = (from q in p.Roles 
                                    select new Role()
                                    {
                                        Id = q.RoleId, 
                                        Name = q.Role.Name
                                    }).ToArray()
                    });

            return await result.ToArrayAsync(ctoken);
        }
    }
}
